code/game/objects/items/weapons/grenades/smokebomb.dm

```diff
@@ -37,3 +35,7 @@
 	sleep(80)
 	qdel(src)
 	return
+
+/obj/item/weapon/grenade/smokebomb/attackby(obj/item/I as obj, mob/user as mob)
+	if(istype(I,/obj/item/device/multitool))
+		smoke_color = input(user, "Choose a color for the smoke:") as color|null
```

Comment:
No, not particularly. If the user cancels the color selection should remain as whatever the smoke_color var holds.